lib/graphql/batch/executor.rb

```diff
@@ -0,0 +1,38 @@
+module GraphQL::Batch
+  class Executor
+    THREAD_KEY = :"#{name}.batched_queries"
+    private_constant :THREAD_KEY
+
+    def self.current
+      Thread.current[THREAD_KEY] ||= new
+    end
+
+    attr_reader :loaders
+
+    def initialize
+      @loaders = {}
+    end
+
+    def shift
+      @loaders.shift.last
+    end
+
+    def execute(loader)
+      loader.execute
+    rescue => exc
+      loader.reject_all(exc)
```

Comment:
this now rejects even the promises that had been fulfilled prior to the exception... is that intended or does `promise.rb` make that a no-op so we don't care?